dot/state/epoch.go

```diff
@@ -884,12 +1006,62 @@ func (s *EpochState) FinalizeBABENextEpochData(finalizedHeader *types.Header) er
 	for e := range s.nextEpochData {
 		if e <= nextEpoch {
 			delete(s.nextEpochData, e)
+			// remove the epoch data from the database
+			if err = s.deleteEpochDataFromDisk(e); err != nil {
+				return fmt.Errorf("cannot delete next epoch data from the database: %w", err)
+			}
 		}
 	}
 
 	return nil
 }
 
+func (s *EpochState) deleteEpochDataFromDisk(epoch uint64) error {
+	configKeysToDelete, err := getDataKeysFromDisk[types.NextEpochData](s.db, nextEpochDataPrefix, epoch)
+	if err != nil {
+		return fmt.Errorf("cannot get next config data keys from disk: %w", err)
+	}
+
+	for _, key := range configKeysToDelete {
+		err = s.db.NewBatch().Del([]byte(key))
+		if err != nil {
+			return fmt.Errorf("cannot delete next config data from the database: %w", err)
+		}
+	}
+	return nil
+}
+
+// getDataKeysFromDisk is a generic function that returns all the nextEpochData or  nextConfigData keys
+// for a given epoch from the database
+func getDataKeysFromDisk[T types.NextEpochData | types.NextConfigDataV1](
+	db GetterPutterNewBatcherPrefixIter, prefix []byte, currentEpoch uint64) (
+	[]string, error) {
+
+	dataKeys := []string{}
+
+	iter, err := db.NewPrefixIterator(prefix)
+	if err != nil {
+		return dataKeys, err
+	}
+
+	defer iter.Release()
+
+	for iter.First(); iter.Valid(); iter.Next() {
+		_, epoch, fork, err := getNextEpochOrConfigData[T](iter)
+		if err != nil {
+			return dataKeys, err
+		}
+		// if the epoch is the current epoch, then we append the key to the dataKeys
+		if epoch == currentEpoch {
+			key := append(prefix, []byte(fmt.Sprintf("-%d:%s", epoch, fork))...)
+			dataKeys = append(dataKeys, string(key))
+		}
```

Comment:
since you already has the iterator with the keys, you dont need to call `getNextEpochOrConfigData`, you just need to get the `iter.Key()` and append to the set of keys to delete
```suggestion
		dataKeys = append(dataKeys, string(iter.Key()))
```